update(docs): revamped and expanded documentation (#12)
cross-seed/cross-seed.org: docs/reference/faq-troubleshooting.md

# FAQ & Troubleshooting

### What can I do about `error parsing torrent at http://…`?

This means that the jacket download link didn't resolve to a torrent file. It's
possible you got rate-limited so you might want to try again in a day or more.
Otherwise, just ignore it. There's nothing cross-seed will be able to do to fix
it.

### rtorrent injected torrents don't check (or start at all) until force rechecked

Remove any `stop_untied=` schedules from your .rtorrent.rc.

Commonly: `schedule2 = untied_directory, 5, 5, (cat,"stop_untied=",(cfg.watch),"*.torrent")`

### Failed to inject, saving instead.

Best way to start troubleshooting this is to check the `verbose.\*.log` and find this specific event.

You will be able to see the circumstances around the failure, and start investigating why this occurred.

### My data-based results are paused at 100% after injection

This is by design and due to the way data-based searches function. This is done to prevent downloading missing files

> or files that failed rechecking automatically. Until there is a way to guarentee you won't end up downloading

Reviewer: -> "prevent automatically downloading ... rechecking."

extra (not cross-seeding) this is the best solution.

> Consider using infoHash if you are racing using data-based searches to prevent mismatches.

Reviewer: infoHash uses the .torrent matching pipeline, not data-based. I like the recommendation for infoHash if they're racing but the second half of the sentence should go

Author: Yea I don't know where that came from, I wrote it at 4AM....so who knows.

### My data-based search is searching torrent files!

Torrent files in torrentDir take precedence over data files with the same name and are deduplicated. If you need to search
specifically the dataDir, use an empty directory for torrentDir temporarily. This will strictly search the data.

### Error: ENOENT: no such file or directory

> This is usually caused by broken symlinks in your dataDir during a data-based search. Check the `/logs/verbose.\*.log` files for the file causing this.

Reviewer: or files moving/getting deleted mid-search

Author: forgot that, I'm going to add a note here as well about not linking files inside your dataDir.

Author: This is preventable if you do not link files within your `dataDir`, or have them outside of the `maxDataDepth` visibility.

Sounds a bit clunky, any thoughts?

### What linkType should I use? (data-based searching)

Your options are `hardlink` or `symlink`. These operate in seperate ways, and depending on your workflow you should choose appropriately. This is a brief description, however
a more in depth guide is available at [Trash's Hardlinking Guide](https://trash-guides.info/Hardlinks/Hardlinks-and-Instant-Moves/).

- symlinks are a "shortcut" of a sort, pointing at the original file from a new location. This can be used across mounts (docker) or partitions/drives. The issue is that if the original file is deleted when you remove a torrent, the torrents in your client using the symlink will "break" and you will receive errors. If this sounds like a hassle, consider reading further about hardlinks.

> 

Reviewer: also discuss symlink advantages (broken symlinks take no space) and disadvantages of hardlinks (broken links still take up space)

also plug qbm for hardlinks I think, since that really is the optimal setup imo

Author: That's a benefit for hardlinks for me :)

Covering it in next commit.

- hardlinks are a tricky thing for someone not familiar, but are worth understanding. They are a "direct line" to the actual data on the disk and are indistinguishable from the original to the file system. They do not require any extra space. These linktypes are only possible on the same partition/disk and you will need to have your linkDir set to the same mount (docker) or partition. This is the best approach if you do not keep torrents in your client (due to them being deleted from the tracker) - which would then break symlinks and cross-seeds. This is the approach commonly used in Arr's on import.

### Searches aren't working from VPN'd torrent client

If you are running your searches on completion from a VPN'd torrent client, it is likely you will need to set up split tunneling. You will not be able to access your local instance (which should not be publicly exposed) from the VPN due to localhost/NAT not being accessible from the VPN. If you are running a VPN'd torrent client in docker, you can use your cross-seed's container IP or name (if using custom docker networks.)

### Searching media libraries vs. torrent data (data-based searching)

> You can search both your media libraries (Arr/Plex) and actual torrent data (downloaded files.) If you are using the media libraries, you will need to use `matchMode: "risky"` in your configuration file to allow cross-seed some leeway in its matching process for these renamed files. `risky` `matchMode` is not recommended to be used without skipRecheck set to false, as this is not as accurate as safe.

Reviewer: ```suggestion
You can search both your media libraries (Arr/Plex) and actual torrent data (downloaded files). If you are using the media libraries with renamed files, you will need to use `matchMode: "risky"` in your configuration file to allow cross-seed some leeway in its matching process. `risky` `matchMode` is not recommended to be used without skipRecheck set to false, as it may result in more false positives than `safe`.
```

> - Due to the way data-based searching works, with tv media libraries cross-seed has no way of knowing whether the season is complete. Therefore it will not be able to match full seasons, even if the season is from a season pack.

Reviewer: ```suggestion
- Due to the way data-based searching works, risky matching only matches renamed files if they are a single-file torrent. As TV libraries are often renamed, data-based matching will not be able to pick up matches on these files.
```

### My season packs are cross-seeding individual episodes!

> There is a configuration option, `includeSingleEpisodes`, which expands from `includeEpisodes`. If you wish to search for season pack's as a whole torrent and invidual episodes NOT from a season pack, you will need to set `includeEpisodes` to false, and `includeSingleEpisodes` to true. Both options would be best utilized with includeNonVideos set to true.

Reviewer: ```suggestion
You can use `includeSingleEpisodes`, which expands from `includeEpisodes`. If you wish to search for season packs as a whole and individual episodes _not_ from a season pack, you will need to set `includeEpisodes` to false, and `includeSingleEpisodes` to true. Both options would be best utilized with includeNonVideos set to true.
```

### Why do I see "filetree is different" in my logs?

> This is due to the current matching algorithm cross-seed uses, and is most commonly associated with extra .nfo/.srt file's being present in a torrent or the possible case of one torrent being in a folder and the potential match not being (or vice versa.)

Reviewer 2: could recommend running the `cross-seed diff` command here

> This is due to the current matching algorithm cross-seed uses, and is most commonly associated with extra .nfo/.srt file's being present in a torrent or the possible case of one torrent being in a folder and the potential match not being (or vice versa.)

Reviewer: this result also happens when a file simply has a different name between the candidate and searchee, even if the "tree" is the same (e.g. both single file torrents)

Author: This is due to the current matching algorithm cross-seed uses, and is most commonly associated with extra .nfo/.srt file's being present in a torrent, the possible case of one torrent being in a folder and the potential match not being (or vice versa), or filenames within torrents not matching.

?

> ### My tracker is mad at me for downloading 1000000 .torrent files!

Reviewer 2: ```suggestion
### My tracker is mad at me for snatching 1000000 .torrent files!
```

> cross-seed currently searches for potential matches and compares the contents and size of files within, it is possible you are running searches too often, are not keeping the torrent_cache folder, or are improperly utilizing [**autobrr**](https://autobrr.com/).

Reviewer: add a caveat that while we try to reduce grabs, to compare the files of two torrents we simply have to grab the torrent and some grabs are unavoidable

### Recommended Overall Improvements

>     - You can use the `announce` end point instead of `webhook` to match what cross-seed already knows about your available media.

Reviewer 2: idk where these "end point" mentions came from (ik they were here before you) but it is and always has been "endpoint"

```suggestion
    - You can use the `announce` endpoint instead of `webhook` to match what cross-seed already knows about your available media.
```

Author: I added this entire section, so that was me. I think Grammarly did that. Yeah....we'll go with Grammarly. :P
